src/SW_Output_get_functions.c

```diff
@@ -169,34 +197,45 @@ void get_co2effects_text(OutPeriod pd, SW_ALL *sw) {
             OUT_DIGITS,
             sw->VegProd.veg[k].co2_multipliers[WUE_INDEX][simyear]
         );
-        strcat(go->sw_outstr, str);
+        strcat(OutRun->sw_outstr, str);
     }
 }
 #endif
 
 #if defined(RSOILWAT) || defined(SWNETCDF)
-void get_co2effects_mem(OutPeriod pd, SW_ALL *sw) {
+void get_co2effects_mem(OutPeriod pd, SW_RUN *sw, SW_OUT_DOM *OutDom) {
     int k;
     size_t iOUTIndex = 0;
-    SW_GEN_OUT *go = &sw->GenOutput;
-    RealD *p = go->p_OUT[eSW_CO2Effects][pd];
+    SW_OUT_RUN *OutRun = &sw->OutRun;
+    RealD *p = OutRun->p_OUT[eSW_CO2Effects][pd];
 
 #if defined(RSOILWAT)
     get_outvalleader(
-        &sw->Model, pd, go->irow_OUT, go->nrow_OUT, go->tOffset, p
+        &sw->Model,
+        pd,
+        OutRun->irow_OUT,
+        OutDom->nrow_OUT,
+        sw->OutRun.tOffset,

```

Comment:
Since we define `*OutRun`, we could do `OutRun->tOffset` instead of `sw->OutRun.tOffset`